cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -348,6 +348,119 @@ pub struct TypedManifestCatalog {
     pub(super) options: ManifestOptions,
 }
 
+impl TypedManifestCatalog {
+    pub fn pkg_descriptor_with_id(&self, id: impl AsRef<str>) -> Option<ManifestPackageDescriptor> {
+        pkg_descriptor_with_id(id, &self.install.0)
+    }
+
+    pub fn pkg_descriptors_in_toplevel_group(&self) -> Vec<(String, ManifestPackageDescriptor)> {
+        pkg_descriptors_in_toplevel_group(&self.install.0)
+    }
+
+    pub fn pkg_descriptors_in_named_group(
+        &self,
+        name: impl AsRef<str>,
+    ) -> Vec<(String, ManifestPackageDescriptor)> {
+        pkg_descriptors_in_named_group(name, &self.install.0)
+    }
+
+    pub fn pkg_or_group_found_in_manifest(&self, name: impl AsRef<str>) -> bool {
+        pkg_or_group_found_in_manifest(name.as_ref(), &self.install.0)
+    }
+
+    pub fn pkg_belongs_to_non_empty_named_group(
+        &self,
+        pkg: impl AsRef<str>,
+    ) -> Result<Option<String>, ManifestError> {
+        pkg_belongs_to_non_empty_named_group(pkg.as_ref(), &self.install.0)
+    }
+
+    pub fn pkg_belongs_to_non_empty_toplevel_group(
+        &self,
+        pkg: impl AsRef<str>,
+    ) -> Result<bool, ManifestError> {
+        pkg_belongs_to_non_empty_toplevel_group(pkg.as_ref(), &self.install.0)
+    }
```

Comment:
Opinion piece for the sake of time feel free to ingore!




I feel like most of this could have been implemented on a `Map<GroupName, Set<package_id>>` structure without having to iterate the install table over and over and adding 12 methods/functions that are only used once eventually.

Reply:
I went this route because I think adding lookup functionality is generally useful for the manifest and it's easy enough to iterate through maps rather than creating a new data type.